File: Tarea1Unidad3Python/Cliente.py
import datetime
import random

class Cliente:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad
        self.num_pedido = random.randint(1000, 9999)
        self.id = self.generar_id()
        self.productos_comprados = []
        self.compras = []
        self.fecha_pedido = datetime.date.today()
        self.compra_hecha = False

    def get_nombre(self):
        return self.nombre

    def get_id(self):
        return self.id

    def get_edad(self):
        return self.edad

    def get_numero_pedido(self):
        return self.num_pedido

    def get_fecha_pedido(self):
        return self.fecha_pedido

    def is_compra_hecha(self):
        return self.compra_hecha

    def set_compra_hecha(self, compra_hecha):
        self.compra_hecha = compra_hecha

    def get_productos_comprados(self, i):
        return str(self.compras[i])

    def generar_id(self):
        letra_nombre = self.get_nombre()[0]
        pedido = str(self.get_numero_pedido())
        self.set_id("C" + letra_nombre + "-" + pedido)
        return self.id

    def set_id(self, id):
        self.id = id

    def get_datos(self):
        print(f"ID: {self.get_id()}\nNombre: {self.get_nombre()}\nEdad: {self.get_edad()}\nNúmero de Pedido: {self.get_numero_pedido()}\nFecha del pedido: {self.get_fecha_pedido()}\nCompra realizada: {self.is_compra_hecha()}")

    def pedido(self, producto):
        self.productos_comprados.append(producto)

File: Tarea1Unidad3Java/src/Cliente/Cliente.java
package Cliente;

import Productos.Producto;

import java.time.LocalDate;
import java.util.*;
public class Cliente {
    private String nombre;
    private String id;
    private int edad;
    private int numPedido;
    private ArrayList<Producto> productosComprados;
    private ArrayList<Compras> compras;
    private LocalDate fechaPedido;
    private boolean compraHecha;


    public Cliente(String nombre, int edad) {
        this.nombre = nombre;
        this.edad=edad;
        this.numPedido= rd.nextInt(1000, 9999);
        this.id=generateID();
        this.productosComprados = new ArrayList<>();
        this.compras = new ArrayList<>();
        this.fechaPedido = LocalDate.now();
        this.compraHecha=false;
    }

    public String getNombre() {
        return nombre;
    }
    public String getId() {
        return id;
    }
    public int getEdad() {
        return edad;
    }
    public int getNumeroPedido() {
        return numPedido;
    }
    public LocalDate getFechaPedido() {
        return fechaPedido;
    }
    public boolean isCompraHecha() {
        return compraHecha;
    }
    public void setCompraHecha(boolean compraHecha) {
        this.compraHecha = compraHecha;
    }
    public String getProductosComprados(int i) {
        return (compras.get(i)).toString();
    }

    Random rd = new Random();
    public String generateID(){
        char letraNombre = getNombre().charAt(0);
        String pedido = String.valueOf(getNumeroPedido());
        setId("C"+letraNombre+("-")+pedido);
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public void getDatos(){
        System.out.printf("\nID: %s \nNombre: %s \nEdad: %s \nNúmero de Pedido: %s \nFecha del pedido: %s \nCompra realizada: %b", getId(), getNombre(), getEdad(), getNumeroPedido(), getFechaPedido(), isCompraHecha());
    }
    public void pedido(Producto producto){
        productosComprados.add(producto);
    }
}